### ISSUE TITLE
GUI request - "Synchronization involving a FAT partition" option to include "NTFS" as well

### ISSUE THREAD

**Issue by xandark** [bcpierce00/unison#591]
When creating a profile within the GUI on the "Specific Options" page, it says:

```Synchronization involving a FAT partition```

This tripped me up when I was trying to create a profile to sync from an EXT4 partition to an NTFS partition. Because it said "FAT", I didn't enable that feature and wound up getting errors on the initial sync regarding xattr and thought it was impossible. After thinking about it awhile, I recreated a profile but this time enabled the "Synchronization involving a FAT partition" option, even though it was an NTFS partition.

Could the GUI change the wording to be more clear? Like...

```Synchronization involving a FAT/NTFS partitions```

**Comment by gdt** [comment]
Pull requests welcome of course.
I wonder if that's the issue, or if it's really "file system not supporting xattr", which if so is not entirely about windows.

I am guessing you are syncing files that do have extended attributes.  It strikes me as a separate and a bug if syncing files that do not involve extended attributes to a fs that does not support them has issues.

**Comment by xandark** [comment]
Reflecting back on it, I do not remember the exact error message it said, maybe it was something to do with chmod. See, I'm grasping at the moment. Regardless, if the string were more inclusive, I would've dodged the issue.

**Comment by gdt** [comment]
Please figure out what's really going and why and update this issue.

**Comment by xandark** [comment]
I didn't realize you were waiting on me to test this further, but I will do so.

**Comment by gdt** [comment]
It's not so much waiting, as until someone can really clearly explain what's wrong, and then produce a change to be merged, and argue convincingly that the change is correct, there will be no progress.    I am functioning more as someone who decides if proposed changes are safe, than someone who tries to fix things that are reported.

**Comment by xandark** [comment]
Okay, I ran some tests with Unison 2.48.4 on my Linux KDE Neon laptop, which mirrors Ubuntu 20.04 for system libs.

I created four partitions: FAT32, EXFAT, NTFS, and HFS+.

Each one would be synchronized with two files from a F2FS partition. One of the two files had an xattr, one did not.

I created two sync profiles for each partition: one with the "Synchronization involving a FAT partition" option disabled, and one enabled.

What I found is that the only partition of the four that would sync with the "Synchronization involving a FAT partition" option disabled was the HFS+ partition. 

The other partitions, FAT32, EXFAT, and NTFS, all failed to sync any file if the "Synchronization involving a FAT partition" option was disabled. Only by changing the option to enabled would the files synchronize.

In all successful sync cases, the file with the xattr lost the xattr after the sync to these four partition types. But that's to be expected, and I only mention it because it was speculated that that was the cause of the failure to sync on NTFS in the initial report.

A representative sync error message is:

```
Donna.png — transport failure
• Failed to set permissions of file /media/michael/NTFS/.unison.Donna.png.1afcc15842f58063b730a478b9af367f.unison.tmp to rw-rw-r--: the permissions was set to rwxrwxrwx instead. The filesystem probably does not support all permission bits. If this is a FAT filesystem, you should set the "fat" option to true. Otherwise, you should probably set the "perms" option to 0o1664 (or to 0 if you don't need to synchronize permissions).
```

So to conclude, I'd like to amend my initial suggestion that the GUI change its wording to:

`Synchronization involving a FAT, EXFAT, or NTFS type partition`

I'd also recommend changing the transport failure textual string to say: "If this is a FAT, EXFAT, or NTFS filesystem, ..."

**Comment by gdt** [comment]
It strikes me that the real bug is that we're talking about "FAT" when we should be talking about something like 'filesystem that does not support permissions".  And perhaps the right thing should just happen.

Thanks for digging in.  Now that we understand, I've tried to summarize the bug crisply in a new issue #638, which references this as background -- so I'm closing this as a dup.    It, like many others, awaits a well-thought-out pull request :-)
